core/controllers/admin.py

```diff
@@ -205,6 +265,46 @@ def _reload_collection(self, collection_id):
             raise Exception('Cannot reload a collection in production.')
 
 
+class AdminRoleHandler(base.BaseHandler):
+    """Handler for roles tab of admin page. Used to view and update roles."""
+
+    @require_super_admin
+    def get(self):
+        view_method = self.request.params['method']
+
+        if view_method == 'role':
+            role = self.request.params['role']
+            users_by_role = {
+                username: role
+                for username in user_services.get_usernames_by_role(role)
+            }
+            self.render_json(users_by_role)
+        elif view_method == 'username':
+            username = self.request.params['username']
+            userid = user_services.get_user_id_from_username(username)
+            if userid is None:
+                self.render_json({})
```

Comment:
Is this an error case? I.e. should this return an error because the client shouldn't request for userid = None?